src/preamble.js

```diff
@@ -2223,6 +2223,7 @@ function integrateWasmJS(Module) {
       exports = instance.exports;
       if (exports.memory) mergeMemory(exports.memory);
       Module['asm'] = exports;
+      asm = exports;
```

Comment:
Sounds good, done in https://github.com/kripken/emscripten/pull/5099/commits/fdcedf47f0d303727d6727eee0ac136b3a7e7d9e

Ideally we could scope the global `asm` variable to be local to instantiation. Alternatively we can avoid declaring `var asm` at all, and just use `Module['asm']` directly.
##
... ok tried that in 50bed1839513b5627beba9d7c6e6b93d6a87123f which seems to work, what do people think?
Reasons I like it:
* removes the ability to get use `asm` wrongly, forcing use of `Module['asm']`

Reasons we should probably revert that commit:
* it's more different from asm.js (if we want to do this we can do so for asm.js as well)
* any non-Emscripten js libraries that used the old `asm` will break